sn_networking/src/event/swarm.rs

```diff
@@ -659,6 +674,30 @@ impl SwarmDriver {
                 .set(self.swarm.connected_peers().count() as i64);
         }
     }
+
+    /// Insert into the latest connected peers list. This list does not contain all the connected peers.
+    /// Older entries are removed if the list exceeds 50.
+    fn insert_latest_connected_peers(&mut self, addr: Multiaddr) {
+        while self.latest_connected_peers.len() >= 50 {
+            // remove the oldest entry
+            let Some(oldest) = self
+                .latest_connected_peers
+                .iter()
+                .min_by_key(|(_, time)| *time)
+                .map(|(addr, _)| addr.clone())
+            else {
```

Comment:
this `else` branch is bit weird, on what condition, there will `break` happens ?